zerver/decorator.py

```diff
@@ -125,8 +125,8 @@ def get_client_name(request, is_json_view):
         else:
              return "Unspecified"
 
-def process_client(request, user_profile, is_json_view=False):
-    client_name = get_client_name(request, is_json_view)
+def process_client(request, user_profile, is_json_view=False, client_name=None):
+    client_name = get_client_name(request, is_json_view) if client_name is None else client_name
```

Comment:
let's do this using multiple lines -- we don't usually use the inline `if` statement style since it is less readable.